File: kakao/19년 겨울 인턴/인형뽑기.cpp
#include <bits/stdc++.h>
using namespace std;

int solution(vector<vector<int>> board, vector<int> moves)
{
    int answer=0;
    stack<int> basket;

    for(int i=0;i<moves.size();i++)
    {
        int check=moves[i]-1;
        for(int j=0;j<board.size();j++)
        {
            if(board[j][check])
            {
                if(!basket.empty()&&basket.top()==board[j][check])
                {
                    basket.pop();
                    answer+=2;
                }
                else
                    basket.push(board[j][check]);
                board[j][check]=0;
                break;
            }
        }
    }
    return answer;
}

File: kakao/19년 겨울 인턴/인형뽑기.py
def solution(board, moves):
    basket=[0];answer=0
    for j in moves:
        for i in range(len(board)):
            if board[i][j-1]:
                basket.append(board[i][j-1])
                board[i][j-1]=0
                if basket[-1]==basket[-2]:
                    basket.pop()
                    basket.pop()
                    answer+=2     
                break
    return answer